versioning/BRM/extentmap.cpp

```diff
@@ -2080,10 +2092,12 @@ void ExtentMap::grabEMIndex(OPS op)
     }
     else
     {
+      const bool isReadOnly = false;
       // Sending down current Managed Shmem size. If EMIndexImpl instance size doesn't match
       // fEMIndexShminfo->allocdSize makeExtentMapIndexImpl will remap managed shmem segment.
       fPExtMapIndexImpl_ =
-          ExtentMapIndexImpl::makeExtentMapIndexImpl(getInitialEMIndexShmkey(), fEMIndexShminfo->allocdSize);
+          ExtentMapIndexImpl::makeExtentMapIndexImpl(getInitialEMIndexShmkey(), fEMIndexShminfo->allocdSize,
+                                                     isReadOnly, emIndexLocked, op == READ ? &fMST : nullptr);
 
       if (r_only)
         fPExtMapIndexImpl_->makeReadOnly();
```

Comment:
`makeReadOnly()` should be under WRITE lock.

Reply:
JFYI you are mixing locks up. mutex lock that is taken for the scope of the function controls multithread races whilst shmem RWLock controls interprocess races.  fPExtMapIndexImpl_ is process-specific entity.

Reply:
`makeReadOnly()` is calling `BRMManagedShmImpl::remap()` under the hood, which is a write op to the shmem segment.